File: retort-annotations/src/main/java/co/kaioru/retort/annotation/IAnnotationFactory.java
package co.kaioru.retort.annotation;

import co.kaioru.retort.ICommand;
import co.kaioru.retort.ICommandContext;

import java.util.List;

public interface IAnnotationFactory<I extends ICommandContext, O> {

    List<IAnnotationAdapter<I, O>> getPipeline();

    List<ICommand<I, O>> generate(Object object);

}

File: retort-annotations/src/main/kotlin/co/kaioru/retort/annotation/IAnnotationFactory.kt
package co.kaioru.retort.annotation

import co.kaioru.retort.ICommand

interface IAnnotationFactory<I, O> {
    val adapters: MutableCollection<IAnnotationAdapter<I, O>>

    fun registerAdapter(adapter: IAnnotationAdapter<I, O>) {
        this.adapters.add(adapter)
    }

    fun registerAdapters(adapters: Collection<IAnnotationAdapter<I, O>>) {
        this.adapters.addAll(adapters)
    }

    fun generate(input: Any): Collection<ICommand<I, O>>
}